src/CabanaPD_Input.hpp

```diff
@@ -268,6 +276,22 @@ class Inputs
     bool contains( std::string label ) { return inputs.contains( label ); }
 
   protected:
+    void compareCriticalTimeStep( std::string name, double dt, double sum )
+    {
+        double safety_factor = inputs["timestep_safety_factor"]["value"];
+        double rho = inputs["density"]["value"];
+        double dt_crit = safety_factor * std::sqrt( 2 * rho / sum );
```

Comment:
This is only correct for mechanics, which is:
![Uploading Mechanics.jpg…]()

Reply:
I just copied your implementation originally, thanks for the details